packages/core/src/sync-store/encoding.ts

```diff
@@ -1,98 +1,112 @@
-      transactionIndex: MAX_CHECKPOINT.transactionIndex,
-      eventType: EVENT_TYPES.blocks,
-      eventIndex: ZERO_CHECKPOINT.eventIndex,
-    }),
-    baseFeePerGas: block.baseFeePerGas
-      ? hexToBigInt(block.baseFeePerGas)
-      : null,
-    difficulty: hexToBigInt(block.difficulty),
-    number: hexToBigInt(block.number),
-    timestamp: hexToBigInt(block.timestamp),
-    extraData: block.extraData,
-    gasLimit: hexToBigInt(block.gasLimit),
-    gasUsed: hexToBigInt(block.gasUsed),
-    logsBloom: block.logsBloom!,
-    miner: toLowerCase(block.miner),
-    mixHash: block.mixHash ?? null,
-    nonce: block.nonce ?? null,
-    parentHash: block.parentHash,
-    receiptsRoot: block.receiptsRoot,
-    sha3Uncles: block.sha3Uncles ?? null,
-    size: hexToBigInt(block.size),
-    stateRoot: block.stateRoot,
-    totalDifficulty: block.totalDifficulty
-      ? hexToBigInt(block.totalDifficulty)
-      : null,
-    transactionsRoot: block.transactionsRoot,
-  };
-};
+}: { block: SyncBlock; chainId: number }): Insertable<BlocksTable> => ({
+  chain_id: chainId,
+  number: hexToNumber(block.number),
+  timestamp: hexToNumber(block.timestamp),
+  hash: block.hash,
+  parent_hash: block.parentHash,
+  logs_bloom: block.logsBloom!,
+  miner: toLowerCase(block.miner),
+  gas_used: hexToBigInt(block.gasUsed),
+  gas_limit: hexToBigInt(block.gasLimit),
+  base_fee_per_gas: block.baseFeePerGas
+    ? hexToBigInt(block.baseFeePerGas)
+    : null,
+  nonce: block.nonce ?? null,
+  mix_hash: block.mixHash ?? null,
+  state_root: block.stateRoot,
+  receipts_root: block.receiptsRoot,
+  transactions_root: block.transactionsRoot,
+  sha3_uncles: block.sha3Uncles ?? null,
+  size: hexToBigInt(block.size),
+  difficulty: hexToBigInt(block.difficulty),
+  total_difficulty: block.totalDifficulty
+    ? hexToBigInt(block.totalDifficulty)
+    : null,
+  extra_data: block.extraData,
+});
+
+export const decodeBlock = ({ block }: { block: DbBlock }): InternalBlock => ({
+  number: BigInt(block.number),
+  timestamp: BigInt(block.timestamp),
+  hash: block.hash,
+  parentHash: block.parent_hash,
+  logsBloom: block.logs_bloom,
+  miner: block.miner,
+  gasUsed: BigInt(block.gas_used),
+  gasLimit: BigInt(block.gas_limit),
+  baseFeePerGas: block.base_fee_per_gas ? BigInt(block.base_fee_per_gas) : null,
+  nonce: block.nonce,
+  mixHash: block.mix_hash,
+  stateRoot: block.state_root,
+  receiptsRoot: block.receipts_root,
+  transactionsRoot: block.transactions_root,
+  sha3Uncles: block.sha3_uncles,
+  size: BigInt(block.size),
+  difficulty: BigInt(block.difficulty),
+  totalDifficulty: block.total_difficulty
+    ? BigInt(block.total_difficulty)
+    : null,
+  extraData: block.extra_data,
+});
 
 type LogsTable = {
-  id: string;
-  chainId: number;
-  checkpoint: string | null;
-  blockHash: Hash;
-  blockNumber: ColumnType<string, string | bigint, string | bigint>;
-  logIndex: number;
-  transactionHash: Hash;
```

Comment:
Seems a bit risky to not store transaction and block hash on logs. We've definitely used them for debugging RPC provider errors and reorg bugs.